upgrade/steps.go

```diff
@@ -261,6 +262,39 @@ func UpgradeProviderVersion(
 	return step.Combined("Update TF Provider", steps...)
 }
 
+var maintenanceRelease = stepv2.Func10E("Check if we should release a maintenance patch", func(
+	ctx context.Context,
+	repo ProviderRepo,
+) error {
```

Comment:
I agree and this is what I did at first; however [the rest of this codebase does not](https://github.com/pulumi/upgrade-provider/blob/bb4318b313e32435d964af6292b4a72506fc1e50/upgrade/steps.go#L1016) so I've followed that pattern here for consistency.

The test works fine; at least insofar as testing `ctx.MaintenancePatch`.

Reply:
The rest of the codebase uses `ctx` as an implicit return because it was written against `step` (v1) which did not allow returning values.

We are trying to migrate over to a style of explicit return values.